src/core/document.js

```diff
@@ -744,7 +742,7 @@ class PDFDocument {
       IsLinearized: !!this.linearization,
       IsAcroFormPresent: !!this.acroForm,
       IsXFAPresent: !!this.xfa,
-      IsCollectionPresent: !!this.collection,
+      IsCollectionPresent: this._hasCollection,
```

Comment:
With the changes to the `Catalog.collection` getter, this can be simplified to *only* the following and `this._hasCollection` removed since it's unused (in later commits, you're also removing the temporarily added `this._hasAcroForm` boolean):
```suggestion
      IsCollectionPresent: !!this.catalog.collection,
```